packages/@ngtools/benchmark/src/utils.ts

```diff
@@ -0,0 +1,177 @@
+import * as fs from 'fs';
+import * as child_process from 'child_process';
+const stripAnsi = require('strip-ansi');
+const treeKill = require('tree-kill');
+
+import { BenchmarkOptions } from './benchmark-options';
+
+// Log to console and logFile if it exists
+export function benchmarkLog(benchmarkOptions: BenchmarkOptions, str: string) {
+  console.log(str);
+  if (benchmarkOptions.logFile) {
+    fs.appendFileSync(benchmarkOptions.logFile, str + '\n');
+  }
+}
+
+// Determine is a value is a number
+export function isNumber(value: any) {
+  return !isNaN(parseFloat(value)) && isFinite(value);
+}
+
+// Returns average of all numbers in `arr`
+export function average(arr: any[]) {
+  return arr.reduce((prev, curr) => prev + Number(curr), 0) / arr.length;
+}
+
+// Returns an array of `length` length, filled with `undefined`
+export function makeArray(length: number) {
+  return Array.apply(null, Array(length));
+}
+
+// Make all combinations of array elements
+// http://stackoverflow.com/a/5752056/2116927
+export function combine(array: any[]) {
+  let fn = function (n: number, src: any[], got: any[], all: any[]) {
+    if (n == 0) {
+      if (got.length > 0) {
+        all[all.length] = got;
+      }
+      return;
+    }
+    for (let j = 0; j < src.length; j++) {
+      fn(n - 1, src.slice(j + 1), got.concat([src[j]]), all);
+    }
+    return;
+  };
+  let all: any[] = [];
+  for (let i = 0; i < array.length; i++) {
+    fn(i, array, [], all);
+  }
+  if (array.length > 0) { all.push(array); }
+  return all;
+}
+
+// Returns a promise with the result of calling `times` times the `fn` function with `args`
+// `fn` will be called in parallel, and is expected to return a promise
+// Not used currently
+export function parallelMultiPromise(times: number, fn: any, ...args: string[]) {
```

Comment:
I had it initially but there was no command that could benefit from being ran in parallel. I'll remove it.